### DISCUSSION TITLE
@vercel/kv operations not working when called inside of a callback inside of an Edge function.

### DISCUSSION THREAD

**Discussion by koopuluri** [vercel/community#2525] [Bug Report]
I'm using Vercel Edge function to stream ChatGPT (i.e.Openai /chatCompletion). I pass a callback to the ReadableStream that is called once the stream is complete, which logs the chat in KV.

But any operation on kv in that onComplete() callback freezes. There's no error in logs, nor is it hitting the catch statement. Nothing after it gets executed. 

And interestingly, this ONLY happens in production. Everything works perfectly in my local environment. I've made sure the KV is connected to all environments of the project.

I fixed by moving the kv operations outside of the callback (so that it's executed before and outside of the ReadableStream processing).

This was a headache and there was no mention of this issue anywhere so wanted to share here in case someone else is going through this, and to call this out for the Vercel team.

If I'm doing something fundamentally wrong that led to this, would greatly appreciate some feedback :). Thanks.

**Comment by suss5** [comment]
Im have problems too, maybe more basic question:
Simple KV example build fails:
_Module not found: Can't resolve '@vercel/kv_
Any ideas why that might be?

Its very simple code, like the kv demo code:
Below is the app/locate/route.js file, and I did add kv to the package.json dependencies

import { kv } from "@vercel/kv";
import { NextResponse } from 'next/server';

export async function GET() {
  var num=await kv.incr('views')
  return NextResponse.json({ number: num});
}

And here is where I call the function
const option = { method: 'GET', }  
 fetch("/locate",option )
     .then(response => response.json())
     .then(response => {  ct1.fillText(response.number,30,30)   } )
                   
Update. Oddly, I changed export to 'export default' and it seems to build now, although it still doesnt work, doesnt show value.
Is the function supposed to go under app directory, or components directory?
Does that make any difference

**Comment by vvo** [comment]
Hey @koopuluri, thanks for the feedback, could you create a small reproduction example showing the error? That would help us a lot, thanks!
